openwisp_notifications/static/openwisp-notifications/js/notifications.js

```diff
@@ -11,9 +12,56 @@ if (typeof gettext === 'undefined') {
         notificationWidget($);
         initNotificationDropDown($);
         initWebSockets($);
+        addToWindowList();
     });
 })(django.jQuery);
 
+window.addEventListener("beforeunload", function () {
+    removeFromWindowList();
+});
+
+function addToWindowList() {
+    // add window id to 'windows' in localStorage
+    windowId = String(Date.now());
+    let tempWindowsList = getWindowList();
+    let index = tempWindowsList.indexOf(windowId);
+    if (index < 0){
+        tempWindowsList.push(windowId);
+    }
+    localStorage.windows = tempWindowsList.join(',');
+}
+
+function removeFromWindowList() {
+    // remove window id from 'windows' in localStorage on reload and close
+    var tempWindowsList = getWindowList();
+    let index = tempWindowsList.indexOf(String(windowId));
+    if (index > -1){
+        tempWindowsList.splice(index, 1);
+    }
+    localStorage.windows = tempWindowsList.join(",");
+}
+
+function getWindowList() {
+    // get 'windows' list in from of array
+    var tempWindowsList = [];
+    if (localStorage.windows){
+        tempWindowsList = localStorage.windows.split(",");
+    }
+    return tempWindowsList;
+}
+
+function canPlaySound() {
+    // give access to play sound to first window of the list
+    let tempWindowsList = getWindowList();
+    let access_to = tempWindowsList[0];
+    if (access_to === windowId){
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
```

Comment:
these functions could be rewritten in a more compact form:

```javascript
const notificationWindow = {
    add: function(){
        // code
    },
    remove: function(){
        // code
    },
    list: function(){
        // code
    },
    canPlaySound: function(){
        // code
    }
}
```

Less pollution of the global variable namespace.

Reply:
Hi @nemesisdesign , I saw your comments after doing my last amend.  I also like the way you suggested and I did the same. But it is giving error during tests again. In my last amend I had declared `canPlaySound` in `addEventListener` which is in `initWebSockets` function and then tests were passing. What do you think about it?

Reply:
what error is it giving?

Reply:
I tested it again now it is not giving any error. I have made changes as you have suggested. Please review it once again.